CONTRIBUTING.md

```diff
@@ -0,0 +1,662 @@
+```
+
+(This has been edited down to only the relevant parts.)
+
+To start a testbed, each of which is named in the format
+`testbed:«module»:«testcase»`, run the following command:
+
+```console
+user@host:~$ nix run .#testbed:«module»:«testcase»
+```
+
+Any package with a name not fitting the given format is not a testbed, and may
+behave differently with this command, or not work at all.
+
+Once the virtual machine starts, a window should open, similar to the screenshot
+below. The contents of the virtual machine will vary depending on the target you
+selected earlier.
+
+![GDM login screen with a dark background color and showing a guest user](testbed_gnome_default_dark.png)
+
+If the testbed includes a login screen, the guest user should log in
+automatically when selected. Depending on the software used, you may still be
+presented with a password prompt - in which case you can leave it blank and
+proceed by pressing enter.
+
+## Development environment
+
+### Developer shell
+
+To enter the developer shell, run:
+
+```sh
+nix develop
+```
+
+To automatically enter the developer shell upon entering the project directory
+with [`direnv`](https://direnv.net), run:
+
+```sh
+direnv allow
+```
+
+### `pre-commit`
+
+The default developer shell leverages [`pre-commit`](https://pre-commit.com)
+hooks to simplify the process of reaching minimum quality standards for casual
+contributors. This means applying code formatters, and scanning for things like
+unused variables which should be removed.
+
+By default, once you have entered the developer shell, `pre-commit` runs
+automatically just before you create a commit. This will only look at the files
+which are about to be committed.
+
+You can also run it manually against all files:
+
+```sh
+pre-commit run --all-files
+```
+
+This is useful if a commit was created outside of the developer shell, and you
+need to apply `pre-commit` to your previous changes.
+
+Note that there is also a flake output, `.#checks.«system».git-hooks`, which
+always runs against all files but does not have access to apply changes. This is
+used in GitHub Actions to ensure that `pre-commit` has been applied.
+
+### `stylix-check`
+
+When a pull request is opened, we use GitHub Actions to build everything under
+`.#checks`. This includes the previously mentioned
+`.#checks.«system».git-hooks`, and every [testbed](./contributing.md##testbeds).
+
+You might sometimes find it useful to run these same checks locally. The built
+in `nix flake check` command does this, however it can be quite slow compared to
+the script we use on GitHub Actions.
+
+To use the same script that we use, you can run this command within the
+developer shell:
+
+```sh
+stylix-check
+```
+
+This is based on [`nix-fast-build`](https://github.com/Mic92/nix-fast-build#readme).
+
+
+### Documentation
+
+The documentation in the `doc` subtree gets
+[published](https://nix-community.github.io/stylix) automatically, using GitHub
```

Comment:
In nixvim, the fix-links wrapper would convert this to a relative link (`./`) when copying into the docs drv.